core/pkg/evaluator/semver.go

```diff
@@ -115,17 +115,26 @@ func parseSemverEvaluationData(values interface{}) (string, string, SemVerOperat
 		return "", "", "", fmt.Errorf("sem_ver evaluation: could not parse operator: %w", err)
 	}
 
-	targetVersion, err := parseSemanticVersion(parsed[2])
+	targetVersion, err := parseSemanticVersion(ensureString(parsed[2]))
 	if err != nil {
 		return "", "", "", fmt.Errorf("sem_ver evaluation: could not parse target value: %w", err)
 	}
 	return actualVersion, targetVersion, operator, nil
 }
 
+func ensureString(v interface{}) string {
+	if str, ok := v.(string); ok {
+		// It's already a string
+		return str
+	}
+	// Convert to string if not already
+	return fmt.Sprintf("%v", v)
+}
+
 func parseSemanticVersion(v interface{}) (string, error) {
 	version, ok := v.(string)
```

Comment:
This should never be possible anymore, and this function is private, so it probably won't happen, but I suppose it's still a good safeguard if it's made public. Up to you if you want to just combine `ensureString` and `parseSemanticVersion` or not.